pkg/controller/egress_router/egress_router_controller.go

```diff
@@ -0,0 +1,173 @@
+package egress_router
+
+// egress router imnplements a controller for the egress router CNI plugin
+
+import (
+	"context"
+	"github.com/openshift/cluster-network-operator/pkg/apply"
+	"github.com/openshift/cluster-network-operator/pkg/render"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/pkg/errors"
+	types2 "go/types"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"log"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"time"
+
+	netopv1 "github.com/openshift/cluster-network-operator/pkg/apis/network/v1"
+	"github.com/openshift/cluster-network-operator/pkg/controller/statusmanager"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Attach control loop to the manager and watch for Egress Router objects
+func Add(mgr manager.Manager, status *statusmanager.StatusManager) error {
+	r, err := newEgressRouterReconciler(mgr, status)
+	if err != nil {
+		return err
+	}
+
+	// Create a new controller
+	c, err := controller.New("egress-router-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource EgressRouter.network.operator.openshift.io/v1
+	err = c.Watch(&source.Kind{Type: &netopv1.EgressRouter{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &EgressRouterReconciler{}
+var manifestDir = "bindata/"
+
+type EgressRouterReconciler struct {
+	mgr    manager.Manager
+	client client.Client
+	status *statusmanager.StatusManager
+
+	egressrouters    map[types.NamespacedName]*egressrouter
+	egressrouterErrs map[types.NamespacedName]error
+}
+
+var ResyncPeriod = 5 * time.Minute
+
+func newEgressRouterReconciler(mgr manager.Manager, status *statusmanager.StatusManager) (reconcile.Reconciler, error) {
+
+	return &EgressRouterReconciler{
+		mgr:    mgr,
+		status: status,
+		client: mgr.GetClient(),
+
+		egressrouters:    map[types.NamespacedName]*egressrouter{},
+		egressrouterErrs: map[types.NamespacedName]error{},
+	}, nil
+}
+
+func (r EgressRouterReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling egressrouter.network.operator.openshift.io %s\n", request.NamespacedName)
+
+	obj := &netopv1.EgressRouter{}
+	err := r.mgr.GetClient().Get(context.TODO(), request.NamespacedName, obj)
+
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Printf("Egress Router %s seems to have been deleted\n", request.NamespacedName)
```

Comment:
If the egress router is deleted, you should clean things up - delete any local references, and (perhaps) delete any kubernetes resources. However, the easiest way to delete kubernetes resources is to set the OwnerReference.